javascript/javascript3/week2/script.js

```diff
@@ -0,0 +1,59 @@
+const selectedCurrency = document.getElementById("my-currency");
+const changeToCurrency = document.getElementById("change-currency");
+const inputTag = document.getElementById("amount");
+const button = document.getElementById("button");
+const result = document.getElementById("result");
+let dataAll;
+let dataKeys;
+// ---- getting the data from API ----
+const getCurrency = async () => {
+  const currencyData = await fetch("https://open.er-api.com/v6/latest/USD");
+  const data = await currencyData.json();
+  dataAll = data.rates;
+  dataKeys = Object.keys(dataAll);
+  currencyRender(dataKeys);
+};
+getCurrency();
+// ---- render the data in the DOM ----
+const currencyRender = (array) => {
+  selectedCurrency.innerHTML = array
+    .map((e) => {
+      if (e === "EUR") {
+        return `<option selected="selected" value="${e}">${e}</option>`;
+      }
+      return `<option value="${e}">${e}</option>`;
+    })
+    .join("");
+  changeToCurrency.innerHTML = array
+    .map((e) => {
+      if (e === "DKK") {
+        return `<option selected="selected" value="${e}">${e}</option>`;
+      }
+      return `<option value="${e}">${e}</option>`;
+    })
+    .join("");
+};
+// ---- calculate the exchange ----
+const calculation = (amount, type1, type2) => {
+  amount = inputTag.value;
+  const converter = (dataAll[type2] * amount) / dataAll[type1];
```

Comment:
It seems like you are converting all the currencies to USD then converting the USD value to the target value, which is actually really intelligent since you only make the one API call. 

While you are clearly very smart, unfortunately it didn't help you here since in most cases you would not be able to get away with making the one API call. In those cases you would need to call the API each time you get the user's input. Just an FYI.